### ISSUE TITLE
Library doesn't work

### ISSUE THREAD

**Issue by Y-Grigoriy** [horchler/DynamixelTools#1]
Hello,

I try to use this library in MATLAB 2017a for control Dynamixel MX-28 via OpenCM9.04. I checked motor ID and serviceability of the microcontroller via Robotics_OpenCM IDE, but in MATLAB i have error everytime. Code and error:
`USBBaudRate = 1000000;
Dxl = DXL(USBBaudRate,'COM8');
DXL.BaudRateValue.BAUD_1000000
ID=7;
Dxl.ping(ID)==ID
Serial connection timeout.

ans =

  logical

   0`
Could you help me?

**Comment by horchler** [comment]
@GrigoriyAY -sorry for the delay. I've not had time to look into your issue. But I do have some questions to help me debug this:

What model of MX-28 is this (-28T, -28R, -28AT, -28AR)?
What model number is it (value at address 0 in EEPROM)? 
What firmware version is it using (value at address 1 in EEPROM)?
Have you been able run either of my demos?
Are you sure that the MX-28 has been configured to use the 1000000 baud rate?
Have you reset the OpenCM9.04 and tried uploading a basic blink program in the ROBOTIS IDE?
Can you command the MX-28 through then ROBOTIS IDE?

**Comment by Y-Grigoriy** [comment]
Sorry, that I didn't answer you early. I had to work with USB2Dynamixel (it works good, but I need Bluetooth connection for control Dynamixel motors).
1. MX-28T
2. 29
3. 39
4. I tried to run your demo, it works in last year, but doesn't work now (I try to reinstall MATLAB and library, that couldn't help)
From Demo I have following error:

Warning: The specified amount of
data was not returned within the
Timeout period.
'serial' unable to read any data.
For more information on possible
reasons, see Serial Read Warnings. 

5.  Yes (I checked all parameters in RoboPlus)
6 & 7. Everything works well in ROBOTIS IDE

**Comment by horchler** [comment]
Hmmmm. Thanks for all of the info. Strange. Since the last time this worked was a while ago, I'm guessing that you may have updated your OS and/or Matlab version or are using a different computer?

 - I assume that you've confirmed that your serial port name is correct? And that you've tried relaunching Matlab and rebooting (sometimes serial ports get locked up)?
 - Can you run the [`OpenCMDemo`](https://github.com/horchler/DynamixelTools/blob/master/demo/OPENCMDemo.m) successfully?
 - Do you have other types of servos that you could try to see if they work?
 - Have you tried factory resetting one of your MX-28T in RoboPlus?
 - Have you updated the firmware of the servos in RoboPlus recently? V39 of the firmware support Robotis Protocol 2.0 which my DynamixelTools haven't been updated to support. I'm pretty sure it also supports Protocol 1.0, but you might consider downgrading the firmware if it's possible (and if you did actually update it).

**Comment by Y-Grigoriy** [comment]
Yes, it's other computer, but I can work only on this at this time. My current MATLAB version is R2018a and I work in Windows 10 with current actual version. Serial port is correct. 
1. OpenCMDemo works successfully.
2. I try MX-106T, the same situation.
3. I made firmware recovery, that couldn't help.
4. Today I updated MX-28T to firmware 40, that couldn't help. I used Protocol 1.0.
